crates/parcel_core/src/asset_graph.rs

```diff
@@ -0,0 +1,345 @@
+          // Propagate re-exported symbol to dependency.
+          re_exports.insert(sym.local.clone());
+        }
+      }
+
+      // Propagate to all export * wildcard dependencies.
+      wildcards.insert(star);
+    } else {
+      // Otherwise, add each of the requested symbols to the asset.
+      for sym in requested_symbols.iter() {
+        if asset_requested_symbols.insert(sym.clone()) {
+          if let Some(asset_symbol) = asset.symbols.iter().find(|s| s.exported == *sym) {
+            if asset_symbol.is_weak {
+              // Propagate re-exported symbol to dependency.
+              re_exports.insert(asset_symbol.local.clone());
+            }
+          } else {
+            // If symbol wasn't found in the asset or a named re-export.
+            // This means the symbol is in one of the export * wildcards, but we don't know
+            // which one yet, so we propagate it to _all_ wildcard dependencies.
+            wildcards.insert(sym.clone());
+          }
+        }
+      }
+    }
+
+    let deps: Vec<_> = self
+      .graph
+      .neighbors_directed(asset_node, Direction::Outgoing)
+      .collect();
+    for dep_node in deps {
+      let dep_index = self.dependency_index(dep_node).unwrap();
+      let DependencyNode {
+        dependency,
+        requested_symbols,
+        state,
+      } = &mut self.dependencies[dep_index];
+
+      let mut updated = false;
+      for sym in &dependency.symbols {
+        if sym.is_weak {
+          // This is a re-export. If it is a wildcard, add all unmatched symbols
+          // to this dependency, otherwise attempt to match a named re-export.
+          if sym.local == "*" {
+            for wildcard in &wildcards {
+              if requested_symbols.insert(wildcard.clone()) {
+                updated = true;
+              }
+            }
+          } else if re_exports.contains(&sym.local)
+            && requested_symbols.insert(sym.exported.clone())
+          {
+            updated = true;
+          }
+        } else if requested_symbols.insert(sym.exported.clone()) {
+          // This is a normal import. Add the requested symbol.
+          updated = true;
+        }
+      }
+
+      // If the dependency was updated, propagate to the target asset if there is one,
+      // or un-defer this dependency so we transform the requested asset.
+      // We must always resolve new dependencies to determine whether they have side effects.
+      if updated || *state == DependencyState::New {
+        if let Some(resolved) = self
+          .graph
+          .edges_directed(dep_node, Direction::Outgoing)
+          .next()
+        {
+          self.propagate_requested_symbols(resolved.target(), dep_node, on_undeferred);
+        } else {
+          on_undeferred(dep_node, Arc::clone(&dependency));
+        }
+      }
+    }
+  }
+}
+
+#[derive(serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SerializedAsset {
+  id: u64,
```

Comment:
Can you make this and the `SerializedDependency` a `String` type and add `.to_string()` in the conversion? It's what breaks the memory fs, since the number is too big for JS

Reply:
We could reduce the size of these numbers ; to other benefits

Since they will never grow this large

Reply:
Yeah, either is fine. Not sure what the motivation Devon had for making it u64 before.